miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/RealMiniAppDisplay.kt

```diff
@@ -0,0 +1,45 @@
+package com.rakuten.tech.mobile.miniapp.display
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import android.widget.FrameLayout
+import androidx.annotation.VisibleForTesting
+import com.rakuten.tech.mobile.miniapp.MiniAppDisplay
+
+@SuppressLint("SetJavaScriptEnabled")
+internal class RealMiniAppDisplay(context: Context, val basePath: String) : MiniAppDisplay,
+    WebView(context) {
+
+    init {
+        setLayoutParams(
+            FrameLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT
+            )
+        )
+        settings.javaScriptEnabled = true
+        settings.allowUniversalAccessFromFileURLs = true
+        webViewClient = MiniAppWebViewClient()
+        loadUrl(getLoadUrl())
+    }
+
+    override fun setWebViewClient(client: WebViewClient?) {
+        super.setWebViewClient(client ?: MiniAppWebViewClient())
+    }
+
+    @VisibleForTesting
+    fun getLoadUrl() = "file://$basePath/index.html"
+
+    override fun getMiniAppView(): View = this
+}
+
+internal class MiniAppWebViewClient : WebViewClient() {
```

Comment:
Do you think this should be private?

Reply:
One of the unit tests is checking this part. Just added annotation to manage the case.